docs/ref/contrib/admin/index.txt

```diff
@@ -1183,7 +1183,8 @@ subclass::
 
     If ``readonly_fields`` is used without defining explicit ordering through
     :attr:`ModelAdmin.fields` or :attr:`ModelAdmin.fieldsets` they will be
-    added last after all editable fields.
+    added last after all editable fields. Otherwise, the read-only field
+    replaces the editable field.
```

Comment:
OK, so I'm yet to open this up in the debugger and trace exactly what's going on but, it looks like this statement isn't true. 

As I read the ticket, you've made it so the read-only field replaces the editable field in the default position **unless** `fields` or `fieldsets` is used. 

Is that correct? 

If so I come back to the question I put on the ticket, why isn't using `fields` or `fieldsets` the right way to go here? 
(Putting read only fields at the end is the established behaviour; it will surprise people if we just change it; and there's a long-standing approach to control ordering if you want it.)

Reply:
Thanks for taking the time to review it. I think the current behavior is as following: If `fields` or `fieldsets` are set explicitly, the readonly fields are shown at the position as defined by these. If they are not set explicitly, the read-only fields are shown at the end. Indeed, the statement is incorrect. I should have changed the paragraph to something like

> A read-only field replaces the corresponding editable field if it appears in `fields` or `fieldsets` either explicitly or in the default value. Otherwise, the read-only field is added at the end.

While I see your point regarding surprising behavior, let explain my motivation. My pull-request changes the above behavior such that in booth cases the read-only fields are shown in order. If I don't set `fields` or `fieldsets`, my expectation is that everything behaves as if I had set `fields` to the default value (more or less the fields of the model in the order they were defined). In order the get the desired effect (default order with read-only fields replacing their editable ones), I would have to restate the default order for `fields`.

Reply:
Wait, I think this is not correct. Let me check a few things, I'll come back later.